### PR TITLE
MRG: shifted onsets of annotations created by mark_flats

### LINKED ISSUE #6295: mark_flats results in annotations that are out of data range
#### Describe the bug
`mne.preprocessing.mark_flats` appends flat segments to `raw.annotations` that are out of data range.

This probably happens because `mne.preprocessing.mark_flats` adds annotations relative to the first sample while `set_annotations` has changed in `0.17` so that `orig_time` matches `info['meas_date']` and not the first sample.

#### Steps and/or code to reproduce

```Python
import numpy as np
import mne

# Data
sfreq = 1000  # Sampling frequency
times = np.arange(0, 10, 0.001)  # Use 10000 samples (10s)

sin = np.sin(times * 10)  # Multiplied by 10 for shorter cycles
flat = np.zeros(20)

data = np.atleast_2d(np.array(np.r_[sin, flat, sin]))

# Info
ch_types = ['mag']
ch_names = ['sin_flat_sin']

info = mne.create_info(ch_names=ch_names, sfreq=sfreq, ch_types=ch_types)
info['meas_date'] = (1545655654, 71165)

# Raw
raw = mne.io.RawArray(data, info, first_samp=50000)

# Mark flats
mne.preprocessing.flat.mark_flat(raw)

# The annotation is omitted though it should not be
raw.set_annotations(raw.annotations)
```

#### Expected results
Nothing changes when I run `raw.set_annotations(raw.annotations)`

#### Actual results
The segment is omitted due to being out of data range.

```
<stdin>:1: RuntimeWarning: Omitted 1 annotation(s) that were outside data range.
```

#### Additional information
```
Platform:      Windows-7-6.1.7601-SP1
Python:        3.7.3 (default, Apr 24 2019, 15:29:51) [MSC v.1915 64 bit (AMD64)
]
Executable:    D:\Windows_software\Miniconda3\envs\mne_dev_test\python.exe
CPU:           Intel64 Family 6 Model 69 Stepping 1, GenuineIntel: 4 cores
Memory:        Unavailable (requires "psutil" package)
mne:           0.18.dev0
numpy:         1.16.3 {blas=mkl_rt, lapack=mkl_rt}
scipy:         1.2.1
matplotlib:    Not found

sklearn:       Not found
nibabel:       Not found
mayavi:        Not found
cupy:          Not found
pandas:        Not found
dipy:          Not found
```

### PR DESCRIPTION
**Pull request by kalenkovich**
Onsets were calculated relative to the first sample and not to the `raw.info['meas_date']`.

#### Reference issue
Fixes #6295.

#### What does this implement/fix?
I shifted the onsets by `raw.first_samp / raw.info['sfreq']` to the right so that their timing is relative to  `raw.info['meas_date']`.

### CODE REVIEW

**Comment by larsoner** [comment]
Is it possible to add some test? For example take a segment of raw FIF data, load the first 3 sec of data, zero out the 1-2 sec interval, run `mark_flat` and show that it fails to mark the correct segment on `master` but does so on this PR?

If it's not clear how to do this I can give it a shot and make a PR into your branch.

**Comment by larsoner** [comment]
Pushed a commit that updates a few tests. Can you modify `whats_new.rst` to the name and URL you prefer? Then we can merge!

File: mne/preprocessing/tests/test_flat.py [hunk 1]
```diff
@@ -10,14 +10,17 @@
...
     """Test marking flat segments."""
     # Test if ECG analysis will work on data that is not preloaded
     n_ch, n_times = 11, 1000
     data = np.random.RandomState(0).randn(n_ch, n_times)
     assert not (np.diff(data, axis=-1) == 0).any()  # nothing flat at first
     info = create_info(n_ch, 1000., 'eeg')
-    raw = RawArray(data, info)
+    info['meas_date'] = (1, 2)
+    # test first_samp != for gh-6295
+    raw = RawArray(data, info, first_samp=first_samp)
```
Comment (larsoner) [inline mne/preprocessing/tests/test_flat.py]: This was the test modification that failed on `master` but passed on this PR.

The ones below for `test_annotations` are just to ensure we stay robust in annotation-related tests that use `RawArray`.

**Comment by kalenkovich** [comment]
@larsoner, sorry for not replying earlier and thank you for updating the test.

I am fine with the name/url you added.

I am curious if this bugfix belongs in `whats_new.rst` since the fix came before the release of `mark_flats`.

**Comment by agramfort** [comment]
thx @kalenkovich and @larsoner

**Comment by larsoner** [comment]
> I am curious if this bugfix belongs in whats_new.rst since the fix came before the release of mark_flats.

This is true, I probably could have instead added your name to the `whats_new.rst` about adding the feature itself, but it's not a big difference either way. I just wanted to make sure that, especially as a new contributor, your name showed up in the release notes for the contribution.

**Comment by kalenkovich** [comment]
Got it @larsoner, thank you.

**Pull request merged**